hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -133,7 +133,8 @@ public static void deleteMetadataTable(String basePath, HoodieEngineContext cont
   public static List<HoodieRecord> convertMetadataToRecords(HoodieCleanMetadata cleanMetadata, String instantTime) {
     List<HoodieRecord> records = new LinkedList<>();
     int[] fileDeleteCount = {0};
-    cleanMetadata.getPartitionMetadata().forEach((partition, partitionMetadata) -> {
+    cleanMetadata.getPartitionMetadata().forEach((partitionName, partitionMetadata) -> {
+      final String partition = partitionName.equals("") ? NON_PARTITIONED_NAME : partitionName;
```

Comment:
I see we are repeating this partition name transformation in 2 other places in this change and few other places in the code. Are there any common transformation that can be done to done partition names to convert all these empty to NON_PARTITIONED_NAME? This way we will not miss out on individual places. 

Like, how about in the update() method for all table operations we examine the partition names and change them if needed?

Reply:
unfortunately I could not find any other good place to fit it in. 
we will have a handle to partition name only by going through the commit metadata (getPartitionMetadata), and we have to fix it before creating HoodieRecord. Let me know if you can find a better way to go about this. happy to incorporate.

Reply:
isn't update() method a right place for this? That is the origin for all convert metadata.